qa/tasks/cephfs/test_volume_client.py

```diff
@@ -995,14 +996,19 @@ def test_put_object_versioned(self):
 
         # Test if put_object_versioned() crosschecks the version of the
         # given object. Being a negative test, an exception is expected.
-        with self.assertRaises(CommandFailedError):
-            self._volume_client_python(vc_mount, dedent("""
-                data, version = vc.get_object_and_version("{pool_name}", "{obj_name}")
-                data += 'm1'
-                vc.put_object("{pool_name}", "{obj_name}", data)
-                data += 'm2'
-                vc.put_object_versioned("{pool_name}", "{obj_name}", data, version)
-            """).format(pool_name=pool_name, obj_name=obj_name))
+        expected_exception = 'rados_OSError'
+        output = self._volume_client_python(vc_mount, dedent("""
+                    data, version = vc.get_object_and_version("{pool_name}", "{obj_name}")
+                    data += 'm1'
+                    vc.put_object("{pool_name}", "{obj_name}", data)
+                    data += 'm2'
+                    try:
+                        vc.put_object_versioned("{pool_name}", "{obj_name}", data, version)
+                    except {expected_exception}:
+                        print('{expected_exception} raised')
```

Comment:
It'll be better if you instead raise a custom exception PutObjectVersionedFailedError, and assert that this error is raised.

Reply:
Add PutObjectVersionedFailedError to this test or ceph_volume_client.py?

Reply:
In the test is fine